### PR TITLE
[1.1.2] Move hostname address registration into separate, conditional test

### LINKED ISSUE #1729: [Patch]: Flaky test on OSX Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.AddressRegistrationTests.RegisterAddresses_IPv4_Success
```C#
System.Net.Sockets.SocketException : Network is unreachable
   at Microsoft.AspNetCore.Testing.HttpClientSlim.<GetSocket>d__7.MoveNext() in /Users/asplab/TeamCity/work/d6337ac2abf63a4b/.repositories/KestrelHttpServer/test/shared/HttpClientSlim.cs:line 114
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Testing.HttpClientSlim.<GetStream>d__6.MoveNext() in /Users/asplab/TeamCity/work/d6337ac2abf63a4b/.repositories/KestrelHttpServer/test/shared/HttpClientSlim.cs:line 85
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Testing.HttpClientSlim.<GetStringAsync>d__1.MoveNext() in /Users/asplab/TeamCity/work/d6337ac2abf63a4b/.repositories/KestrelHttpServer/test/shared/HttpClientSlim.cs:line 24
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.AddressRegistrationTests.<RegisterAddresses_Success>d__7.MoveNext() in /Users/asplab/TeamCity/work/d6337ac2abf63a4b/.repositories/KestrelHttpServer/test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs:line 93
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
```

### PR DESCRIPTION
**Pull request by natemcmaster**
Test changes only. Backports #1751 to the 1.1.2.

Closes #1729.

### CODE REVIEW

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/DnsHostNameIsResolvableAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+        public bool IsMet
+        {
+            get
+            {
+                try
+                {
+                    _hostname = Dns.GetHostName();
+                    var addresses = Dns.GetHostAddressesAsync(_hostname).GetAwaiter().GetResult();
+                    if (addresses.Any(i => !IPAddress.IsLoopback(i)))
```
Comment (halter73) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/DnsHostNameIsResolvableAttribute.cs]: I know this is already in dev, but what's the problem if the hostname resolves to localhost?

Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/DnsHostNameIsResolvableAttribute.cs]: The test is trying to verify that when the server address `http://something` is given, the server is accessible via `http://[::]` or `http://0.0.0.0` and not loopback. We have other tests that verify loopback.

Comment (halter73) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/DnsHostNameIsResolvableAttribute.cs]: We have other tests that verify loopback, but they don't verify that it works when the hostname is specified via the IServerAddresses feature. We know it's the equivalent of "*" and "+" today, but I don't think it hurts to verify.

Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/DnsHostNameIsResolvableAttribute.cs]: In 1.1, do we bind to loopback when hostname is specified? We don't in 2.0

Comment (halter73) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/DnsHostNameIsResolvableAttribute.cs]: The behavior doesn't change. We try to bind to IPv6Any, but request made to localhost/hostname should still succeed.

Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/DnsHostNameIsResolvableAttribute.cs]: Ok. So remove the IsLoopback filter?

Comment (halter73) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/DnsHostNameIsResolvableAttribute.cs]: Yep. The rest LGTM.

**Pull request merged**